graftabledag/graftabledag.go

```diff
@@ -47,7 +49,7 @@ func New(origin *dag.Dag, dagGetter DagGetter) (*GraftedDag, error) {
 func (gd *GraftedDag) getChaintreeDag(ctx context.Context, did string) (*dag.Dag, error) {
 	tip, err := gd.dagGetter.GetTip(ctx, did)
 	if err != nil {
```

Comment:
I do think we should be explicit about the errors. What i was running into in dgit was `hamt.ErrNotFound`, which will be returned for [server](https://github.com/quorumcontrol/tupelo/blob/78e6495685970f57c5cd1aea5fea5e90f79063be/sdk/gossip/types/graftedownership.go#L108) and [client](https://github.com/quorumcontrol/tupelo/blob/78e6495685970f57c5cd1aea5fea5e90f79063be/sdk/gossip/types/graftedownership.go#L52).

Maybe this can be left up to the `dagGetter`s though, maybe they just return a known error (like `TipNotFound`) and then let it propagate here?

Reply:
Good suggestions. Making it so.

Reply:
That's pushed here and to the corresponding tupelo PR too: https://github.com/quorumcontrol/tupelo/pull/474